### ISSUE TITLE
Opening a File from Nautilus doesn't work sometimes

### ISSUE THREAD

**Issue by cookiengineer** [gnome-mpv/gnome-mpv#391]
Overview Description:

Sometimes, when networked at least 50% of the time, gnome-mpv does not open a file via `nautilus` or `gio`. This means that when double-clicking a file and starting the process, `gnome-mpv` process is started with the correct parameters but all UI elements are blurred out and not changed/updated afterwards.

It's pretty annoying when you have to open files at least 3 times again when it fails with sshfs/gvfs mounted directories more often than usual. It can also be reproduced with SSD drives beyond 500MB/s, so it's not a bandwidth specific problem but seemingly rather an asynchronous missing-to-update-state problem when something is faster or slower than expected in the opening process.

Steps to Reproduce:

1. Use sshfs to reproduce more often, mount an external drive from another machine in gvfs or nautilus / "Other locations" (doesn't matter).
2. Double Click an mkv movie file (happens more often with this video container format).
3. If it happens, all UI elements are greyed out and you have to restart gnome-mpv again.

Version:

- HEAD (of around 20mins ago) on Arch Linux
- 0.15-1 on Arch Linux
- happens on all my 64Bit machines, from core2duo to i7 16-cores, from old HDDs to SSDs with 800MB/s and more, so it's not a bug caused by performance differences.

Additional Information:

I was taking an uneducated look at the codebase, and it seems that the `idle_active` or `ready` flag are updated in a wrong manner: https://github.com/gnome-mpv/gnome-mpv/blob/master/src/gmpv_player.c#L514

In order to trace down more efficiently, I would need more help in debugging. Like in the sense how to log it to stdout and stuff (as I'm a glib/gobject noob).

**Comment by TingPing** [comment]
Will just say I've also been experiencing this for local files (`ntfs-3g` mount) and it has been a relatively recent change.

**Comment by gnome-mpv** [comment]
>Sometimes, when networked at least 50% of the time, gnome-mpv does not open a file via nautilus or gio. This means that when double-clicking a file and starting the process, gnome-mpv process is started with the correct parameters but all UI elements are blurred out and not changed/updated afterwards.

I can't reproduce this (tried opening about 20 mkv files over sftp). Sounds like this might be related to #377, but the bug no longer occurs to me since the fix was applied.

>In order to trace down more efficiently, I would need more help in debugging. Like in the sense how to log it to stdout and stuff (as I'm a glib/gobject noob).

You can enable logging by setting the environment variable `G_MESSAGES_DEBUG=all`. Log messages will be printed to stderr.

**Comment by cookiengineer** [comment]
I wanted to report back here. I'm actually still trying to reproduce the behaviour, but it seems when I activate debugging it doesn't happen anymore and the player controls are blended in always!?

It also does not happen again when I set G_MESSAGES_DEBUG to the default and use a simple `printf()` to stdout. I will continue to test and report back when it happens again.

For now I'm totally confused why it behaves this way and I'm not sure whether it could also be related to a third-party library that I updated by coincidence during the last couple days.

**Comment by Urgau** [comment]
Hello, I would like to say that I have the same problem on my machine and it seems to be related to the version 0.29 of mpv because with version 0.28 no problem.
I also try with G_MESSAGES_DEBUG=all but no change in the output.

**Comment by Urgau** [comment]
Any update ?

**Comment by cookiengineer** [comment]
Nope, can't reproduce on HEAD on my machine once I changed the source code with debugging statements. If I fallback to the AUR build with the current version (`0.15-1`) it happens all the time, so it's at least reproducible with the release before the introduced fix.

The times on how often it happens is reduced, so the current HEAD in its original state will lead to it happening, but only on very rare occasions (when starting to watch 20 movies it happens maybe 1 time) and it feels like it's related to gvfs synchronization problems because memory swaps a lot then.

But once I introduce some debugging statements in the HEAD version from git, I have no idea why it resolves the issue and why it doesn't happen anymore.

Sorry I think my knowledge is not enough to be more efficient in debugging. On my machines it currently only happens on a very old laptop I use for watching series (Dell Latitude with core2duo with not enough jiffies) and my connected machines here with SSHFS. I think it's not CPU or computation power related so if it is a timeout-related problem it is more likely related with the filesystem and/or bandwidth for transfer, and might rather be related to a "slow SSD" or "slow memory" on my old laptop or so.

A fix could be that when it fails to load a file, but knows that there's a file path available and set (via process arguments) that it will not autoplay, but still offer the play button and blend in the UI. Then, UX-wise, the problem would be solved because when I manually click the button it's probably with enough time being passed so that the file playback works then. In comparison; currently the situation always leads to me closing gnome-mpv and opening it again via nautilus; and I guess that this could leave gvfs with enough time to fill up some caches.

**Comment by gnome-mpv** [comment]
>A fix could be that when it fails to load a file, but knows that there's a file path available and set (via process arguments) that it will not autoplay, but still offer the play button and blend in the UI.

When opening a file from Nautilus, the file path is sent to `gnome-mpv` in the form of a DBus message, so we can't use command line arguments here.

**Comment by SyXbiT** [comment]
This is happening for me also. It has been happening for a couple months.
Also on Arch (Currently 4.20 kernel).
I'm using the Nvidia binary driver, but this happens regardless of my mpv.config file (both hardware and software decoding).

When running from the CLI, when the problem occurs, I get the following line in stdin
`Failed to run mpv command "keyup Meta+Super_L". Reason: error running command`
